pkg/registry/populator.go

```diff
@@ -63,7 +63,55 @@ func (i *DirectoryPopulator) Populate(mode Mode) error {
 	return nil
 }
 
+func (i *DirectoryPopulator) globalSanityCheck(imagesToAdd []*ImageInput) error {
+	var errs []error
+	images := make(map[string]bool)
+	for _, image := range imagesToAdd {
+		images[image.bundle.BundleImage] = true
+	}
+
+	for _, image := range imagesToAdd {
+		bundlePaths, err := i.querier.GetBundlePathsForPackage(context.TODO(), image.bundle.Package)
+		if err != nil {
+			// Assume that this means that the bundle is empty
+			// Or that this is the first time the package is loaded.
+			return nil
+		}
+		isExactImage := false
+		for _, bundlePath := range bundlePaths {
+			if _, ok := images[bundlePath]; ok {
+				isExactImage = true
+			}
+		}
+		for _, channel := range image.bundle.Channels {
+			bundle, err := i.querier.GetBundle(context.TODO(), image.bundle.Package, channel, image.bundle.csv.GetName())
+			if err != nil {
+				// Assume that if we can not find a bundle for the package, channel and or CSV Name that this is safe to add
+				continue
+			}
+			if bundle != nil && !isExactImage {
```

Comment:
This wouldn't really need to be per image though, right? Even in disparate channels, you can't add the same bundle to the registry twice.

Reply:
meaning we should exit early once we find the same bundle path on line 83?

Reply:
Yeah, I think if `isExactImage` is set to true you should just return the error for that entire bundle and move to the next one, no need to iterate over the channels at all.

Reply:
Do you think that we could also break out of the channel check loop once we know that one is failing?